alacritty/src/input.rs

```diff
@@ -366,23 +366,66 @@ impl<'a, T: EventListener, A: ActionContext<T>> Processor<'a, T, A> {
         }
     }
 
-    pub fn normal_mouse_report(&mut self, button: u8) {
+    pub fn utf8_mouse_report(&mut self, button: u8) {
         let (line, column) = (self.ctx.mouse().line, self.ctx.mouse().column);
 
-        if line < Line(223) && column < Column(223) {
-            let msg = vec![
+        if line < Line(2015) && column < Column(2015) {
+            let mouse_pos_encode = |pos: usize| -> (u8, u8) {
+                let pos = 32 + 1 + pos;
+                let first = 0xC0 + (pos >> 6);
+                let second =  0x80 + (pos & 0x3F);
+                (first as u8, second as u8)
+            };
+
+            let mut msg = vec![
                 b'\x1b',
                 b'[',
                 b'M',
                 32 + button,
-                32 + 1 + column.0 as u8,
-                32 + 1 + line.0 as u8,
             ];
 
+            if column < Column(95) {
+                msg.push(32 + 1 + column.0 as u8);
+            } else {
+                let encoded_col = mouse_pos_encode(column.0);
+                msg.push(encoded_col.0);
+                msg.push(encoded_col.1);
+            }
+
+            if line < Line(95) {
+                msg.push(32 + 1 + line.0 as u8);
+            } else {
+                let encoded_line = mouse_pos_encode(line.0);
+                msg.push(encoded_line.0);
+                msg.push(encoded_line.1);
+            }
+
             self.ctx.write_to_pty(msg);
         }
     }
 
+    pub fn normal_mouse_report(&mut self, button: u8) {
+        if self.ctx.terminal().mode().contains(TermMode::UTF8_MOUSE) {
```

Comment:
I think we shouldn't have the utf8 mouse handling in `normal_mouse_report`, that is extremely confusing. Instead it should be moved to the `mouse_report` function just like all the other mouse reports.

That way `normal_mouse_report` shouldn't change at all, which makes a lot of sense too.